### PR TITLE
Uncommitted GC prepare include tracked physical addresses

### LINKED ISSUE #4562: Add copied objects information from ref-store
Add new table to graveler KV tracking copied objects.
SaveGarbageCollectionUncommitted should read from this table **after** reading all uncommitted data and add the data to the list of uncommitted objects.

This is part of: https://github.com/treeverse/lakeFS/issues/4475

### PR DESCRIPTION
**Pull request by nopcoder**
Close #4562

### CODE REVIEW

**Review by itaiad200 — CHANGES_REQUESTED** [review]
Can we avoid the double pagination?

File: pkg/catalog/gc_write_uncommitted.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+			continue
+		}
+		// Skip non-relative that address outside the storage namespace
+		entryAddress := entry.Address
+		if entry.Entry.AddressType != Entry_RELATIVE {
+			if !strings.HasPrefix(entry.Address, normalizedStorageNamespace) {
+				continue
+			}
+			entryAddress = entryAddress[len(normalizedStorageNamespace):]
+		}
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply pkg/catalog/gc_write_uncommitted.go:70]: I think this is support for old staging where we added full address to storage namespace.
Didn't modified the code - just extracted this method.

File: pkg/catalog/gc_write_uncommitted.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+	// write uncommitted data from branches
+	var hasData bool
+	if mark == nil || !mark.Tracked {
+		mark, hasData, err = gcWriteUncommittedBranches(ctx, store, pw, repository, w, mark, runID, maxFileSize)
+		if err != nil {
+			return nil, false, err
+		}
+		// we like to return in case data was written and fallthrough to tracked physical addresses if not
+		if hasData {
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply pkg/catalog/gc_write_uncommitted.go]: This check remove empty pagination - accept that we can do better and continue to tracked links also for the case of final chunk.

**Review by itaiad200 — APPROVED** [review]
LGTM

**Review by itaiad200 — APPROVED** [review]

**Pull request merged**